api/components/com_users/src/Controller/UsersController.php

```
@@ -65,4 +87,91 @@ protected function save($recordKey = null)
 
 		return parent::save($recordKey);
 	}
+
+	/**
+	 * User list view with filtering of data
+	 *
+	 * @return  static  A BaseController object to support chaining.
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public function displayList()
+	{
+		$apiFilterInfo = $this->input->get('filter', [], 'array');
+		$filter        = InputFilter::getInstance();
+
+		if (array_key_exists('state', $apiFilterInfo))
+		{
+			$this->modelState->set('filter.state', $filter->clean($apiFilterInfo['state'], 'INT'));
+		}
+
+		if (array_key_exists('active', $apiFilterInfo))
+		{
+			$this->modelState->set('filter.active', $filter->clean($apiFilterInfo['active'], 'INT'));
+		}
+
+		if (array_key_exists('groupid', $apiFilterInfo))
+		{
+			$this->modelState->set('filter.group_id', $filter->clean($apiFilterInfo['groupid'], 'INT'));
+		}
+
+		if (array_key_exists('search', $apiFilterInfo))
+		{
+			$this->modelState->set('filter.search', $filter->clean($apiFilterInfo['search'], 'STRING'));
+		}
+
+		if (array_key_exists('registrationdate', $apiFilterInfo))
+		{
+			$rangeFilter = $filter->clean($apiFilterInfo['registrationdate'], 'STRING');
+
+			if (!in_array($rangeFilter, $this->supportedRange))
+			{
+				// Send the error response
+				$this->sendResponse('registrationdate', 400);
```

This will not be a valid JSON API response. You need to throw an InvalidParameterException from the library https://github.com/tobyzerner/json-api-php/blob/master/src/Exception/InvalidParameterException.php

---

should be correct now 
```json
{
  "errors": [
    {
      "title": "Invalid field: lastvisitdate"
    }
  ]
}
```

---

added the error code 400